src/beliefpropagation.jl

```diff
@@ -0,0 +1,244 @@
+#Construct the random initial Message Tensors for an ITensor Network, based on a partitioning into subgraphs specified ny 'sub graphs'
+#The ITensorNetwork needs to be flat (i.e. just sites and link indices, no site indices), and is assumed to only have 1 link indice between any two sites
+#id_init = 0 => Random, id_init = 1 => Identity Matrix initialisation (preferred)
+function construct_initial_mts(
+  g::NamedDimGraph, flatpsi::ITensorNetwork, dg_subgraphs::DataGraph; id_init=1
+)
+  mts = Dict{Pair,ITensor}()
+
+  for i in vertices(dg_subgraphs)
+    tns_to_contract = ITensor[]
+    for j in neighbors(dg_subgraphs, i)
+      edge_inds = []
+
+      for vertex in dg_subgraphs[i]
+        psiv = flatpsi[vertex]
+        for e in NamedDimEdge.(Ref(vertex) .=> neighbors(g, vertex))
+          if (find_subgraph(dst(e), dg_subgraphs) == j)
+            edge_ind = commoninds(flatpsi, e)[1]
+            push!(edge_inds, edge_ind)
+          end
+        end
+      end
+      if (id_init == 1)
+        A = Array(delta(edge_inds), edge_inds)
+        X1 = ITensor(A, edge_inds)
+      else
+        X1 = randomITensor(edge_inds)
+      end
+      normalize!(X1)
+      mts[i => j] = X1
+    end
+  end
+
+  return mts
+end
+
+#DO a single update of a message tensor using the current subgraph and the incoming mts
+function updatemt(
+  g::NamedDimGraph, flatpsi::ITensorNetwork, subgraph::Vector{Tuple}, mts::Vector{ITensor}
+)
+  Contract_list = ITensor[]
+
+  for m in mts
+    push!(Contract_list, m)
+  end
+
+  for v in subgraph
+    psiv = flatpsi[v]
+    push!(Contract_list, psiv)
+  end
+
+  M = ITensors.contract(Contract_list)
```

Comment:
I guess here we want to use some contraction sequence optimization.

Reply:
Is ITensors not automatically using some optimization (https://docs.juliahub.com/ITensors/P3pqL/0.2.3/ContractionSequenceOptimization.html#ITensors.NDTensors.contract)? Or do I have to enable this with ITensors.enable_contraction_sequence_optimization() ?

Reply:
I would design it where the user passes a keyword argument with a function that computes the contraction sequence, and then the user can decide on the particular algorithm.

For example:
```julia
function updatemt(
  g::NamedDimGraph, flatpsi::ITensorNetwork, subgraph::Vector{Tuple}, mts::Vector{ITensor}; contraction_sequence::Function=tn -> ITensorNetworks.contraction_sequence(tn; alg="optimal")
)
  # [...]
  M = contract(Contract_list; sequence=contraction_sequence(Contract_list))
end
```

Reply:
Good idea, I have put this in